src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx

```diff
@@ -67,13 +73,73 @@ export default class Summary extends React.Component {
     });
   }
 
-  onChangeInterval(e, item) {
-    this.setState({autoReloadInterval: item.key});
+  showApplicationSummary() {
+    const {jobInfo} = this.props;
+    this.showEditor('Application Summary', {
+      language: 'text',
+      value: jobInfo.jobStatus.appExitDiagnostics || '',
+    });
+  }
+
+  showJobConfig() {
+    const {jobConfig} = this.props;
+    this.showEditor('Job Config', {
+      language: 'json',
+      value: JSON.stringify(jobConfig, null, 2),
+    });
+  }
+
+  renderHintMessage() {
+    const {jobInfo} = this.props;
+    if (!jobInfo) {
+      return;
+    }
+
+    const state = getHumanizedJobStateString(jobInfo);
+    if (state === 'Failed') {
+      const diag = jobInfo.jobStatus.appExitDiagnostics;
+      const code = jobInfo.jobStatus.appExitCode;
+      if (code === 177) {
+        // user code error
+        let userExitCode;
+        if (diag) {
+          let match = diag.match(/<Raw>\[ExitCode\]: (\d+)/);
+          if (match) {
+            userExitCode = parseInt(match[1], 10);
+          }
+        }
+        if (!isNil(userExitCode)) {
+          return `Failed due to user error. Exit Code: ${userExitCode}. Please check stderr and stdout for more information.`;
```

Comment:
**Error Type**: User Error;
**Error code**: 1;
**Error log(s)**: please check Stdout and Stderr for each task.